test/unit/path_utils.js

```diff
@@ -22,6 +22,21 @@
       assert.deepEqual(command, expectedSimplified[index], 'should contain a subset of equivalent commands ' + index);
     });
   });
+  QUnit.test('fabric.util.parsePath test special arcs', function(assert) {
+    assert.ok(typeof fabric.util.parsePath === 'function');
+    assert.ok(typeof fabric.util.makePathSimpler === 'function');
+    // eslint-disable-next-line max-len
+    var pathWithWeirdArc = 'p m19.801 17.771h1.457v-1.84c0-1.918.247-3.442.73-4.57.49-1.134 1.313-1.954 2.474-2.473 1.16-.512 2.75-.774 4.773-.774 3.578 0 5.367.875 5.367 2.629 0 .566-.187 1.055-.562 1.457-.371.406-.817.61-1.325.61-.238 0-.652-.047-1.234-.137a10.56 10.56 0 00-1.484-.133c-1.11 0-1.82.324-2.133.976-.316.653-.473 1.583-.473 2.797v1.457h1.504c2.336 0 3.504.707 3.504 2.114 0 1.004-.309 1.64-.93 1.91-.62.27-1.48.402-2.574.402h-1.504v16.238c0 1.215-.289 2.141-.863 2.778-.578.633-1.324.953-2.234.953-.871 0-1.594-.32-2.168-.953-.578-.637-.868-1.563-.868-2.778v-16.238h-1.683c-.914 0-1.617-.207-2.11-.613-.496-.414-.742-.95-.742-1.61 0-1.468 1.024-2.203 3.078-2.203';
+    var parsed = fabric.util.parsePath(pathWithWeirdArc);
+    var atLeastOneArc = false;
+    parsed.forEach(function(command) {
+      if (command.length > 1 && (command[0] === 'a' || command[0] === 'A')) {
+        assert.deepEqual(command.length, 8, 'Arc in SVG should always have size 8 but had less. Did not parse correctly.');
+        atLeastOneArc = true;
+      }
+    });
+    assert.ok(atLeastOneArc, 'No arcs found from SVG but at least one should have been parsed.');
+  });
```

Comment:
a fine test is:

```suggestion
  QUnit.test('fabric.util.parsePath can parse arcs correctly when no spaces between flags', function(assert) {
    // eslint-disable-next-line max-len
    var pathWithWeirdArc = 'a10.56 10.56 0 00-1.484-.133';
    var expected = ['a', 10.56, 10.56, 0, 0, 0, -1.484, -0.133];
    var parsed = fabric.util.parsePath(pathWithWeirdArc);
    var command = parsed[0];
    assert.deepEqual(command, expected, 'Arc should be parsed correctly.');
  });
```

Reply:
Be careful i did not run that test, so it can be wrong.

Other cases to tests:

`path = 'a10.56 10.56 0 011-.133'` <--- is this in spec?
`path = 'a10.56 10.56 1011-.133'` <--- is this in spec?
`path = 'a10.56 10.56 10 11-.133'` <--- is this in spec?

(gonna have a look if they work)

Reply:
This failed when I tried it. It appears that the problem is fabric.reNum doesn't match rNumber is the parsePath method. I reverted to rNumber and the test passed. I think this is also the cause of the visual test failures.